core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -401,6 +401,7 @@ class TransactionsTest extends IntegrationTestHarness {
     producer1.beginTransaction()
     producer1.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, null, "1", "1", willBeCommitted = false))
     producer1.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic2, null, "3", "3", willBeCommitted = false))
+    producer1.flush()
```

Comment:
Does this test fail without the flush?

Reply:
Yes, the produce request from producer1 may interfere with the producer2 produce request. Then it can have other errors.

Reply:
Hmm. Was this different before because the add partitions calls happened automatically (ie without flush?)
Or is it different because we fence epochs on produce now?

In other words, what change in this PR caused us to need this change?

Reply:
I would like to try to get a better understanding of this interaction just to see if it would be a potential problem. 
Does producer 1 cause errors for producer 2 or is it just producer 1 sees errors that the test isn't expecting.

Reply:
The check fails at the next `assertThrows(classOf[ProducerFencedException], () => producer1.commitTransaction())`

Reply:
Do we not throw the error or is it a different error?

Reply:
InvalidProducerEpochException. I think it is from the addPartition response.

Reply:
Oh this is just happening because we changed the error type. (We don't have a separate add partitions call that would happen first.) I think we can remove the flush and just change the error. (commitTransaction should flush anyway)

Reply:
Sure, I updated the errors. But note that, the error type depends on the timing, whether the producer1's produce requests can finish before producer2.

Reply:
Hmm. I understand what you are saying. Ok, sorry to flip flop again, but I think if we see the test is consistent with the flush we can keep it.